src/postinstall/utils/dependencies.utils.ts

```diff
@@ -0,0 +1,30 @@
+import type { PackageJson } from "@storybook/types";
+import type { ConfigFile } from "@storybook/csf-tools";
+import { SUPPORTED_BUILDERS, SupportedBuilders } from "../types";
+
+export const hasDependency = (
+  packageJson: PackageJson,
+  depName: string
+): boolean =>
+  Boolean(
+    packageJson?.dependencies[depName] || packageJson?.devDependencies[depName]
+  );
+
+export const determineBuilder = (mainConfig: ConfigFile): SupportedBuilders => {
+  const frameworkValue = mainConfig.getFieldValue(["framework"]);
+
+  const framework = (frameworkValue?.name || frameworkValue) as string;
```

Comment:
Could you avoid type casting with a type check?

```suggestion
  const framework = typeof frameworkValue === 'string' ? frameworkValue : frameworkValue?.name;
```